examples/controller-example/deployment.yaml

```diff
@@ -0,0 +1,46 @@
+---
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-example
+  namespace: agones-system
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: controller-example
+  template:
+    metadata:
+      labels:
+        app: controller-example
+    spec:
+      serviceAccountName: default
+      containers:
+        - name: controller-example
+          image: us-docker.pkg.dev/agones-mangalpalli/examples/controller-example:0.1
+          imagePullPolicy: Always
+          env:
+            - name: GAMESERVER_IMAGE
+              value: us-docker.pkg.dev/agones-mangalpalli/examples/simple-game-server:0.27
```

Comment:
What's this env var for?

Reply:
I was referring [this](https://spacelift.io/blog/kubernetes-deployment-yaml) and our crd-client for the deployment.yaml. If it isn't required will remove it.